src/graphnet/data/i3extractor.py

```diff
@@ -327,36 +327,89 @@ def __call__(self, frame, padding_value=-1) -> dict:
-                    "OnlineL2Filter_17": int(
-                        bool(frame["FilterMask"]["OnlineL2Filter_17"])
-                    ),
-                }
-            )
 
-        if is_mc and (not is_noise):
+            if frame_has_key(frame, key="FilterMask"):
+                output["DeepCoreFilter_13"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"],
+                        "DeepCoreFilter_13",
+                        default_value=-1,
+                    )
+                )  # what if frame doesnt have deepcorefilter and it says true (bool(-1)=true)
+                output["CascadeFilter_13"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"],
+                        "CascadeFilter_13",
+                        default_value=-1,
+                    )
+                )
+                output["MuonFilter_13"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"], "MuonFilter_13", default_value=-1
+                    )
+                )
+                output["OnlineL2Filter_17"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"],
+                        "OnlineL2Filter_17",
+                        default_value=-1,
+                    )
+                )
+
+            elif frame_has_key(frame, key="DeepCoreFilter_13"):
+                output["DeepCoreFilter_13"] = int(
+                    bool(frame["DeepCoreFilter_13"])
+                )
+
+            if frame_has_key(frame, key="L3_oscNext_bool"):
+                output.update(
+                    {
+                        "L3_oscNext_bool": turn_bool_into_number(
+                            try_get_key(
+                                frame, "L3_oscNext_bool", default_value=-1
+                            )
+                        ),
+                        "L4_oscNext_bool": turn_bool_into_number(
+                            try_get_key(
+                                frame, "L4_oscNext_bool", default_value=-1
+                            )
+                        ),
+                        "L5_oscNext_bool": turn_bool_into_number(
+                            try_get_key(
+                                frame, "L5_oscNext_bool", default_value=-1
+                            )
+                        ),
+                        "L6_oscNext_bool": turn_bool_into_number(
+                            try_get_key(
+                                frame, "L6_oscNext_bool", default_value=-1
+                            )
+                        ),
+                        "L7_oscNext_bool": turn_bool_into_number(
+                            try_get_key(
+                                frame, "L7_oscNext_bool", default_value=-1
+                            )
+                        ),
+                    }
+                )
```

Comment:
My argument being that if `"L3_oscNext_bool" in frame` we are right to assume that `L{3..7}_oscNext_bool` are all in the frame and this allows us to not make a number of checks. Conversely, if this assumption turns out not to be true, this will throw a `KeyError` which I think it perfectly fine, because then the input data would be a bit funky.
```suggestion
                output.update(
                    {
                        "L3_oscNext_bool": int(frame["L3_oscNext_bool"]),
                        "L4_oscNext_bool": int(frame["L4_oscNext_bool"]),
                        "L5_oscNext_bool": int(frame["L5_oscNext_bool"]),
                        "L6_oscNext_bool": int(frame["L6_oscNext_bool"]),
                        "L7_oscNext_bool": int(frame["L7_oscNext_bool"]),
                    }
                )
```